--- apm/node_modules/atom-package-manager/script/postinstall.sh.orig	2019-05-22 09:30:46.241019000 +0900
+++ apm/node_modules/atom-package-manager/script/postinstall.sh	2019-05-22 09:31:21.125020000 +0900
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/local/bin/bash
 
 set -e
 
--- apm/node_modules/atom-package-manager/bin/apm.orig	1985-10-26 17:15:00.000000000 +0900
+++ apm/node_modules/atom-package-manager/bin/apm	2019-05-22 09:30:31.845768000 +0900
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/local/bin/bash
 
 set -e
 
--- apm/node_modules/atom-package-manager/bin/npm.orig	1985-10-26 17:15:00.000000000 +0900
+++ apm/node_modules/atom-package-manager/bin/npm	2019-05-22 09:30:31.844602000 +0900
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/local/bin/bash
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 export PATH="$SCRIPT_DIR:$PATH"
 
--- apm/node_modules/atom-package-manager/bin/python-interceptor.sh.orig	1985-10-26 17:15:00.000000000 +0900
+++ apm/node_modules/atom-package-manager/bin/python-interceptor.sh	2019-05-22 09:30:31.843380000 +0900
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/local/bin/bash
 
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
@@ -34,9 +34,9 @@
       ARGS+=("--format=safemake.py")
     fi
 
-    exec python "${ARGS[@]}"
+    exec python2.7 "${ARGS[@]}"
     ;;
   *)
-    exec python "$@"
+    exec python2.7 "$@"
     ;;
 esac
