--- script/node_modules/electron-packager/targets.js.orig	2019-05-21 23:42:45.365961000 +0900
+++ script/node_modules/electron-packager/targets.js	2019-05-21 23:43:40.375024000 +0900
@@ -5,9 +5,10 @@
 const semver = require('semver')
 
 const officialArchs = ['ia32', 'x64', 'armv7l', 'arm64', 'mips64el']
-const officialPlatforms = ['darwin', 'linux', 'mas', 'win32']
+const officialPlatforms = ['darwin', 'freebsd', 'linux', 'mas', 'win32']
 const officialPlatformArchCombos = {
   darwin: ['x64'],
+  freebsd: ['x64'],
   linux: ['ia32', 'x64', 'armv7l', 'arm64', 'mips64el'],
   mas: ['x64'],
   win32: ['ia32', 'x64']
@@ -21,6 +22,7 @@
 // Maps to module filename for each platform (lazy-required if used)
 const osModules = {
   darwin: './mac',
+  freebsd: './linux',
   linux: './linux',
   mas: './mac', // map to darwin
   win32: './win32'
